drivers/i2s/i2s_nrfx.c

```diff
@@ -377,21 +400,21 @@ static void data_handler(const struct device *dev,
 static void purge_queue(const struct device *dev, enum i2s_dir dir)
 {
 	struct i2s_nrfx_drv_data *drv_data = dev->data;
-	void *mem_block;
+	struct i2s_buf mem_block;
```

Comment:
Please use `struct i2s_buf buf;` for consistency and also because `mem_block.mem_block` looks weird.